src/passes/strings-encoding/StringEncoding.cpp

```diff
@@ -311,6 +351,14 @@ bool StringEncoding::injectOnStackLoop(BasicBlock& BB, Instruction& I, Use& Op,
     fatalError("Can't find the 'decode' routine");
   }
 
+  // TODO: support ObjC strings as well
+  Value *CastEncPtr = nullptr;
+  if (auto *CE = dyn_cast<ConstantExpr>(EncPtr)) {
+    auto *GV = extractGlobalVariable(CE);
+    assert(GV && GV == &G);
```

Comment:
We could do `assert(extractGlobalVariable(CE) == &G)` here and in the bitcast below use `&G` right? This is just a check. Maybe add a comment as well?

Reply:
Definitely, thanks!

Reply:
@antoniofrighetto Maybe comment was the wrong word. The conventional way to write this would be:
```
assert(extractGlobalVariable(CE) == &G && "Previously extracted global variable need to match");
```